api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts

```diff
@@ -0,0 +1,99 @@
+import { v4 } from 'uuid';
+import { Test } from '@nestjs/testing';
+import { CqrsModule, EventBus, IEvent } from '@nestjs/cqrs';
+import { Either } from 'fp-ts/Either';
+
+import {
+  SpecificationId,
+  ScenarioSpecification,
+  SpecificationActivated,
+} from '../../domain';
+
+import { ActivateCandidateSpecificationHandler } from '../activate-candidate-specification.handler';
+import { ScenarioSpecificationRepo } from '../scenario-specification.repo';
+import {
+  ActivateCandidateSpecification,
+  ActivateError,
+} from '../activate-candidate-specification.command';
+
+import { InMemoryScenarioSpecificationRepo } from './scenario-specification-in-memory.repo';
+
+export const getFixtures = async () => {
+  const scenarioId = v4();
+  const currentCandidateSpecificationId = new SpecificationId(v4());
+  const differentCandidateSpecificationId = new SpecificationId(v4());
+  const events: IEvent[] = [];
+
+  const sandbox = await Test.createTestingModule({
+    imports: [CqrsModule],
+    providers: [
+      ActivateCandidateSpecificationHandler,
+      {
+        provide: ScenarioSpecificationRepo,
+        useClass: InMemoryScenarioSpecificationRepo,
+      },
+    ],
+  }).compile();
+  await sandbox.init();
+
+  const scenarioSpecificationRepo: InMemoryScenarioSpecificationRepo = sandbox.get(
+    ScenarioSpecificationRepo,
+  );
+  const sut = sandbox.get(ActivateCandidateSpecificationHandler);
+  const systemEvents = sandbox.get(EventBus);
+
+  systemEvents.subscribe((event) => events.push(event));
+
+  return {
+    GivenScenarioSpecificationWithCandidateWasCreated: () => {
+      scenarioSpecificationRepo.save(
+        new ScenarioSpecification(
+          scenarioId,
+          undefined,
+          currentCandidateSpecificationId,
+        ),
+      );
+    },
+    WhenActivatingCandidateSpecification: async () =>
+      sut.execute(
+        new ActivateCandidateSpecification(
+          scenarioId,
+          currentCandidateSpecificationId.value,
+        ),
+      ),
+    WhenActivatingAnotherSpecification: async () =>
+      sut.execute(
+        new ActivateCandidateSpecification(
+          scenarioId,
+          differentCandidateSpecificationId.value,
+        ),
+      ),
+    ThenScenarioSpecificationIsSaved: async () => {
+      const scenarioSpecification = await scenarioSpecificationRepo.find(
+        scenarioId,
+      );
+      expect(scenarioSpecification).toBeDefined();
+      expect(scenarioSpecification!.currentActiveSpecification).toEqual(
+        currentCandidateSpecificationId,
+      );
+      expect(
+        scenarioSpecification!.currentCandidateSpecification,
+      ).toBeUndefined();
```

Comment:
it's checking more than only that it's saved
the name of the method is a bit misleading

Reply:
As we are in context of saving scenario specification after activating candidate, it verifies if the saved one changed indeed (i.e. persistence applied new version).